lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1096,14 +1108,41 @@ namespace {
     };
 
     void diagnoseMissingCases(RequiresDefault defaultReason, Space uncovered,
-                              const CaseStmt *unknownCase = nullptr) {
+                              const CaseStmt *unknownCase, const CaseStmt *defaultCase) {
+      assert(!(unknownCase && defaultCase) && "both @unknown default and default should be caught earlier");
       if (!Switch->getLBraceLoc().isValid()) {
         // There is no '{' in the switch statement, which we already diagnosed
         // in the parser. So there's no real body to speak of and it doesn't
         // make sense to emit diagnostics about missing cases.
         return;
       }
+
+      // Decide whether we want an error or a warning.
+      std::optional<decltype(diag::non_exhaustive_switch)> mainDiagType =
+          diag::non_exhaustive_switch;
+
+      bool downgrade = false;
```

Comment:
Nit: you might want to leave it where it was, close to where it is used.

Reply:
I was just co-locating it with the main diag, I had to look around at what we were downgrading when I was looking through the file before